packages/react-native-web/src/modules/ImageLoader/index.js

```diff
@@ -122,9 +122,17 @@ const ImageLoader = {
     id += 1;
     const image = new window.Image();
     image.onerror = onError;
-    image.onload = (e) => {
+    image.onload = (nativeEvent) => {
       // avoid blocking the main thread
-      const onDecode = () => onLoad({ nativeEvent: e });
+      const onDecode = () => {
+        nativeEvent.source = {
+          uri: image.src,
+          width: image.naturalWidth,
+          height: image.naturalHeight
+        };
```

Comment:
Attach `source` to `nativeEvent` so `onLoad` matches the RN Image interface https://reactnative.dev/docs/image#onload